plugin/manager.go

```diff
@@ -289,13 +289,54 @@ func attachToLog(id string) func(libcontainerd.IOPipe) error {
 }
 
 func validatePrivileges(requiredPrivileges, privileges types.PluginPrivileges) error {
-	// todo: make a better function that doesn't check order
-	if !reflect.DeepEqual(privileges, requiredPrivileges) {
+	if !isEqual(requiredPrivileges, privileges, isEqualPrivilege) {
 		return errors.New("incorrect privileges")
 	}
+
 	return nil
 }
 
+func isEqual(arrOne, arrOther interface{}, compare func(x, y interface{}) bool) bool {
+	sliceOne := reflect.ValueOf(arrOne)
+	sliceOther := reflect.ValueOf(arrOther)
```

Comment:
Instead of using reflect here, perhaps make `validatePrivileges` strongly typed, and skip the `reflect` here;

```go
func isEqual(arrOne types.PluginPrivileges, arrOther types.PluginPrivileges, compare func(x types.PluginPrivilege, y types.PluginPrivilege) bool) bool {
	if len(arrOne) != len(arrOther) {
		return false
	}

	bool found
	for i := 0; i < len(arrOne); i++ {
		found = false
		for j := 0; j < len(arrOther); j++ {
			if compare(arrOne[i], arrOther[j]) {
				found = true
				break
			}
		}

		if !found {
			return false
		}
	}

	return true
}
```